### ISSUE TITLE
Rewrite URLs in CSS with given path

### ISSUE THREAD

**Issue by raamdev** [websharks/html-compressor#58]
We need to add a new options to the HTML Compressor (`htmlc_rewrite_css_resource_paths`) that will determine whether or not the HTML Compressor should use a specific hostname (a new option called `htmlc_rewrite_host`) when rewriting static resource paths inside CSS files during compression.

@jaswsinc writes... 
> Referencing this method that needs to be refactored: https://github.com/websharks/html-compressor/blob/150321.1/html-compressor/includes/core.php#L1007

This will allow ZenCache to provide an option that lets site owners who have configured a CDN via the Static CDN Filters to also choose to have static resources referenced inside CSS files be served via the CDN. See https://github.com/websharks/zencache/issues/461

---

## Next Actions

- [ ] Add two new options: `htmlc_rewrite_css_resource_paths` (to enable/disable this functionality) and `htmlc_rewrite_host` (a hostname like `d1v41qemfjie0l.cloudfront.net` to be used when rewriting the static resource paths)
- [ ] Refactor [`resolve_css_relatives()`](https://github.com/websharks/html-compressor/blob/150321.1/html-compressor/includes/core.php#L1007) to also support rewriting the path with `htmlc_rewrite_host` if `htmlc_rewrite_css_resource_paths` is enabled

**Comment by jaswsinc** [comment]
Great! Thank you for opening this.

Just wanted to note that there's an option in the HTML Compressor at present which is currently being used by ZenCache whenever Static CDN Filters are enabled in ZenCache. Altering this URL has almost the same effect as `htmlc_rewrite_host` would have, but I agree that we should have a separate `htmlc_rewrite_host` option for this.

- (string)`cache_dir_url_public` A publicly available URL which leads to your cache_dir_public. If you exclude this option, an automatic detection is used (i.e., a best guess based on cache_dir_public).

**Comment by jaswsinc** [comment]
Quick observation I wanted to note here...

Since the HTML Compressor already resolves all relative paths in CSS into full URLs, if the CSS file itself is being served from the CDN, all that's needed to get this working is for the HTML Compressor to remove the host name; i.e., force the CSS resources to use absolute relative paths—relative to the CDN that is serving the file.

In the current release of the HTML Compressor, if you enable CSS compression (enabled by default), this is already the case; i.e., the CSS compressor forces absolute relative paths, because that removes `http://yourdomain.com` from every link; thereby reducing the file size. Leaving you with something like: `background: url('/absolute/relative/path/to/image.png')`

In short, in the current release of ZenCache + Html Compression, if you also enable CSS compression, this is being performed already. Actually, the HTML Compressor is not mapping URLs inside the CSS to your CDN. However, it _is_ mapping the CSS file itself to your CDN, and then forcing absolute relative paths on all resources within that CSS.

---

So what's actually needed here, is to just force absolute relative paths even if the CSS compression is disabled; so long as `htmlc_rewrite_css_resource_paths` _is_ enabled; i.e., either of these should force all absolute relative paths. Absolute relative paths being better than full URLs that lead to the CDN. No reason for full URLs if the CDN is serving the CSS to begin with.

**Comment by jaswsinc** [comment]
Even simpler, we can just force absolute relative (aka: root-relative) paths in all CSS handled by the HTML Compressor. That will resolve this issue indirectly; i.e., it eliminates the need for an `htmlc_rewrite_css_resource_paths` option key altogether. ~ Referencing: https://github.com/websharks/html-compressor/commit/117f40f2df77549d85b3e598aee723e3c4a09e69
